webdriver/tests/bidi/__init__.py

```diff
@@ -18,15 +19,28 @@ def recursive_compare(expected: Any, actual: Any) -> None:
 
     if type(expected) is dict:
         # Actual dict can have more keys as part of the forwards-compat design.
-        assert expected.keys() <= actual.keys(), \
-            f"Key set should be present: {set(expected.keys()) - set(actual.keys())}"
-        for key in expected.keys():
+        expected_missing_keys = set(filter(
+            lambda k: expected[k] == missing,
+            expected.keys()))
+        expected_present_keys = set(expected.keys() - expected_missing_keys)
+
+        assert expected_present_keys <= actual.keys(), \
+            f"Key set should be present: {expected_present_keys - set(actual.keys())}"
+
+        for key in expected_missing_keys:
+            assert key not in actual, f"Key '{key}' should not be present in {actual}"
+
+        for key in expected_present_keys:
             recursive_compare(expected[key], actual[key])
         return
 
     assert expected == actual
 
 
+def missing() -> None:
+    pass
```

Comment:
Feels odd to have a function here, but maybe that's the usual way in Python to define a symbol?

Reply:
I would gladly replace this approach with whatever you find more suitable here

Reply:
> Thanks Maksim, great to have added coverage here.
> 
> I feel like parametrizing all our current tests with `result_ownership` is a bit too much and will not scale. I appreciate that we want to test as many combinations as possible, but I would rather stick to focused tests for now, and maybe add a few more complete tests once we have a decent individual coverage for all parameters.
> 
> For result_ownership, I think we need to test it in 4 cases:
> 
> * returning a value synchronously
> * returning a value asynchronously
> * throwing synchronously
> * throwing asynchronously
> 
> We can then parametrize those with different return values to test maps, sets, objects etc... but at least it will be very clear what the test is about?
> 
> Maybe consider having those new tests in a result_ownership.py file? Let meknow what you think.

Agree. Having those several cases are enough. Having that, I  added a new test file `script/call_function/result_ownership.py` and updated `script/call_function/call_function.py` expectations.

Reply:
N/A anymore